tests/test__data.py

```diff
@@ -220,6 +220,20 @@ def setUp(self):
         r = f.reduced_chi_squareds()
         self.assertIs(type(r), list)
         self.assertAlmostEqual(r[0], 1.5, 1)
+
+        # Try supplying a callable function
+        def tf(x, a1, a2, a3):
```

Comment:
consider: tf -> trial_function
for space on line 228 the string could be set on the previous line
fit_args = 'a1=-1., a2=0.04, a3=0.00006', or perhaps a better name.